src/Azure/Orleans.ScheduledJobs.AzureStorage/AzureStorageJobShardManager.cs

```diff
@@ -124,6 +129,7 @@ public override async Task<List<IJobShard>> AssignJobShardsAsync(SiloAddress sil
         async Task<bool> TryTakeOwnership(AzureStorageJobShard shard, IDictionary<string, string> metadata, SiloAddress newOwner, CancellationToken ct)
         {
             metadata["Owner"] = newOwner.ToParsableString();
+            metadata["MembershipVersion"] = _clusterMembership.CurrentSnapshot.Version.ToString();
```

Comment:
Maybe we should write Version.Value.ToString() instead, so it's always a `long` instead of sometimes being `"default"`.